### PR TITLE
Fix `current` field getting set to tmc context in client config

### LINKED ISSUE #4091: CurrentServer (`current`) field in ClientConfig should not point to TMC context
**Bug description**

current field in the config.yaml is meant to be used to just include managment-cluster/k8s context and it should not include tmc context

For configuring tmc context CLI should use currentContext.mission-control field.

Also, TMC plugins should be using currentContext.mission-control to determine the active tmc context.

This is required for backward compatibility of kubernetes targeted plugins with new context/target functionality

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- (x) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Expected behavior**

CurrentServer (`current`) field in ClientConfig should not point to TMC context.

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it
- Do not set `current` field to point to TMC context in client config

**Details:**
- `current` field in the config.yaml is meant to be used to just include managment-cluster/k8s context and it should not include tmc context
- For configuring tmc context CLI should use `currentContext.mission-control` field.
- Also, TMC plugins should be using `currentContext.mission-control` to determine the active tmc context.

- This is required for backward compatibility of kubernetes targeted plugins with new context/target functionality

### Which issue(s) this PR fixes

Fixes #4091

### Describe testing done for PR

Tested below commands:
- `tanzu login`
- `tanzu context create` 
- `tanzu context use` 

Above commands should not set the `current` to point to TMC context when creating or setting TMC context and just set it in the `currentContext.mission-control` field.

### Release note

```release-note
Fix `current` field getting set to tmc context in client config (config.yaml)
```

### CODE REVIEW

File: cli/runtime/config/servers.go [hunk 1]
```diff
@@ -226,6 +228,15 @@ func RemoveServer(name string) error {
 }
 
 func setCurrentServer(node *yaml.Node, name string) (persist bool, err error) {
+	s, err := getServer(node, name)
+	if err != nil {
+		return false, err
+	}
+
+	if s.Type != configapi.ManagementClusterServerType {
+		return false, errors.Errorf("cannot set non management-cluster server as current server")
```
Comment (marckhouzam) [inline cli/runtime/config/servers.go:237]: Nice!

**Review by marckhouzam — APPROVED** [review]
Tested and it works beautifully.  LGTM

One question however.  What if we use `tanzu login <tmc cluster>` with an old version of the plugin?  Is this possible?  Because then it will still set the `current` field to the tmc cluster...

**Comment by anujc25** [comment]
>One question however. What if we use tanzu login <tmc cluster> with an old version of the plugin? Is this possible? Because then it will still set the current field to the tmc cluster...

Yes. Ideally, TMC users should always be using the new CLI to use this feature as we are adding support in upcoming release but If for some reason, user uses old login plugin to login to tmc than it will set the `current` field.

**Review by vuil — APPROVED** [review]
lgtm, thanks!

**Pull request merged**